ydb/core/tx/datashard/datashard__read_iterator.cpp

```diff
@@ -1683,6 +1712,7 @@ class TDataShard::TReadOperation : public TOperation, public IReadOperation {
         if (Reader->HasUnreadQueries()) {
             Reader->UpdateState(state, ResultSent);
             if (!state.IsExhausted()) {
```

Comment:
Why don't we need `!wasExhausted` check for else branch?

Reply:
Because iterator hasn't produced any results at this time, they cannot be acked and cannot be changed. Since this is the first result ever the exhausted counter hasn't been incremented yet, even if state was somehow exhausted initially.